src/components/application_manager/test/commands/mobile/register_app_interface_request_test.cc

```diff
@@ -0,0 +1,260 @@
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "utils/data_accessor.h"
+#include "mock_message_helper.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "mock_hmi_capabilities.h"
+#include "mock_resume_ctrl.h"
+#include "utils/custom_string.h"
+#include "utils/lock.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::DoAll;
+
+namespace am = ::application_manager;
+
+using ::utils::SharedPtr;
+using am::commands::MessageSharedPtr;
+using am::commands::RegisterAppInterfaceRequest;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+const hmi_apis::Common_Language::eType kHmiLanguage =
+    hmi_apis::Common_Language::EN_US;
+const mobile_apis::Language::eType kMobileLanguage =
+    mobile_apis::Language::EN_US;
+const std::string kMacAddress = "test_mac_address";
+const std::string kAppId = "test_app_id";
+const std::string kDummyString = "test_string";
+}  // namespace
+
+class RegisterAppInterfaceRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  RegisterAppInterfaceRequestTest()
+      : msg_(CreateMessage())
+      , command_(CreateCommand<RegisterAppInterfaceRequest>(msg_))
+      , app_name("test_app_name")
+      , mock_message_helper_(*am::MockMessageHelper::message_helper_mock()) {}
+
+  MessageSharedPtr msg_;
+  SharedPtr<RegisterAppInterfaceRequest> command_;
+
+  const utils::custom_string::CustomString app_name;
+  sync_primitives::Lock lock_;
+  am::ApplicationSet app_set;
+
+  typename IsNiceMock<policy_test::MockPolicyHandlerInterface,
+                      kMocksAreNice>::Result mock_policy_handler_;
+  typename IsNiceMock<resumption::MockResumeCtrl, kMocksAreNice>::Result
+      mock_resum_crt_;
+  typename IsNiceMock<connection_handler_test::MockConnectionHandler,
+                      kMocksAreNice>::Result mock_connection_handler_;
+  typename IsNiceMock<protocol_handler_test::MockSessionObserver,
+                      kMocksAreNice>::Result mock_session_observer_;
+  typename IsNiceMock<application_manager_test::MockHMICapabilities,
+                      kMocksAreNice>::Result mock_hmi_capabilities_;
+  am::MockMessageHelper& mock_message_helper_;
+};
+
+TEST_F(RegisterAppInterfaceRequestTest, Run_MinimalData_SUCCESS) {
+  (*msg_)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*msg_)[am::strings::msg_params][am::strings::app_id] = kAppId;
+  (*msg_)[am::strings::msg_params][am::strings::app_name] = app_name;
+  (*msg_)[am::strings::msg_params][am::strings::language_desired] =
+      kHmiLanguage;
+  (*msg_)[am::strings::msg_params][am::strings::hmi_display_language_desired] =
+      kHmiLanguage;
+  ON_CALL(app_mngr_, IsHMICooperating()).WillByDefault(Return(true));
```

Comment:
@LevchenkoS You can move all ON_CALL in the separate method and call it in both tests. Then tests will become more readable